HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs

```diff
@@ -50,6 +55,122 @@ public ICollection<Examination> BasicSearch(BasicAppointmentSearchDTO parameters
+
+            ICollection<Examination> shiftedAppointments = new List<Examination>();
+            foreach (Examination examination in unavailableAppointments)
+            {
+                List<Examination> availableAppointments = GetShiftedAppointmentForEmergency(parameters.InitialParameters, examination);
+                availableAppointments = availableAppointments.OrderBy(e => e.DateAndTime).ToList();
+                foreach (Examination appointment in availableAppointments)
+                {
+                    if (IsShiftedAppoinmentAvailable(appointment, shiftedAppointments))
+                    {
+                        shiftedAppointments.Add(appointment);
+                        break;
+                    }
+                }
+            }
+
+            return shiftedAppointments;
+        }
+
+        private bool IsShiftedAppoinmentAvailable(Examination appointment, ICollection<Examination> shiftedAppointments)
+        {
+            foreach(Examination shiftedAppointment in shiftedAppointments)
+            {
+                if (shiftedAppointment.DateAndTime.Equals(appointment.DateAndTime)
+                    && (shiftedAppointment.IdPatientCard == appointment.IdPatientCard
+                        || shiftedAppointment.DoctorJmbg == appointment.DoctorJmbg))
+                    return false;
+            }
+            return true;
+        }
+
+        private List<Examination> GetOnlyAdequateAppointments(List<Examination> unavailableAppointments, AppointmentSearchWithPrioritiesDTO parameters)
+        {
+            foreach (Examination examination in unavailableAppointments.ToList())
+            {
+                if (!_roomService.CheckIfRoomHasRequiredEquipment(examination.IdRoom, parameters.InitialParameters.RequiredEquipmentTypes)
+                    || !IsPatientAvailable(parameters.InitialParameters.PatientCardId, examination.DateAndTime)
+                    || !examination.Doctor.CheckIfDoctorHasSpecialty(parameters.SpecialtyId))
+                    unavailableAppointments.Remove(examination);
+            }
+            return unavailableAppointments;
+        }
+
+        private List<Examination> GetUnavailableAppointments(BasicAppointmentSearchDTO parameters)
+        {
+            return (List<Examination>) _examinationRepository.GetExaminationsForPeriod(parameters.EarliestDateTime, parameters.LatestDateTime);
+        }
+
+        private List<Examination> GetShiftedAppointmentForEmergency(BasicAppointmentSearchDTO parameters, Examination examination)
+        {
+            DateTime startDateTime = parameters.LatestDateTime;
+            DateTime endDateTime = new DateTime(startDateTime.Year, startDateTime.Month, startDateTime.Day, 17, 0, 0);
+            List<int> requiredEquipmentTypes = GetRequiredEquipmentForExamination(examination);            
+
+            for (int i = 1; i <= 13; i++)
```

Comment:
13 is because today+13 days represent 2 weeks. There isn't any specific reason other than that.